gammapy/modeling/models/spectral.py

```diff
@@ -1533,6 +1536,105 @@ def from_region_map(cls, map, **kwargs):
         return cls(energy=energy, values=values, **kwargs)
 
 
+class TemplateNDSpectralModel(SpectralModel):
+    """A model generated from a ND array where extra dimensions define the parameter space.
+
+    For more information see :ref:`templateND-spectral-model`.
+
+    Parameters
+    ----------
+    map : `~gammapy.maps.RegionNDMap`
+        Map template. 
+    meta : dict, optional
+        Meta information, meta['filename'] will be used for serialization
+    interp_kwargs : dict
+        Interpolation keyword arguments passed to `gammapy.maps.Map.interp_by_coord`.
+        Default arguments are {'interp': 'linear', 'fill_value': 0}.
+    """
+
+    tag = ["TemplateNDSpectralModel", "templateND"]
+
+    def __init__(self, map, interp_kwargs=None, meta=None, filename=None):
+        self._map = map.copy()
```

Comment:
Maybe one should check that an energy axis is present.

Reply:
added